src/main_window.py

```diff
@@ -1612,6 +1614,8 @@ def logout(self, action, param):
                 os.system("dbus-send --print-reply --session --dest=org.kde.LogoutPrompt /LogoutPrompt org.kde.LogoutPrompt.promptLogout")
             elif self.win.environment == 'COSMIC (New)':
                 os.system("dbus-send --print-reply --session --dest=com.system76.CosmicSession --type=method_call /com/system76/CosmicSession com.system76.CosmicSession.Exit")
+            elif self.win.environment == 'Hyprland':
+                os.system("hyprctl dispatch exit")
```

Comment:
Unfortunately, the `hyprctl` command doesn't work in Flatpak, so you probably need to use dbus to log out the system, or use other options.

Reply:
oh thanks for the feedback i will do that one later because i do not know how to kill hyprland or logout hyprland with a systemcall

Reply:
commented out and wrote "needs to be fixed but i do not know how to make this call on flatpak via python"